### DISCUSSION TITLE
Questions about pocketbase vertical expansion deployment

### DISCUSSION THREAD

**Discussion by luminecs** [pocketbase/pocketbase#5388] [Q&A] [answered]
Hello everyone, I have been learning about pocketbase recently and learned that it uses sqlite as a database and can only be vertically expanded on a single machine. As a result, I have several questions:

1. Can I use the [`pocketbase update`](https://github.com/pocketbase/pocketbase/discussions/2615#discussioncomment-6056186) command to upgrade the current instance with zero downtime? (I don't know its internal working principle at present.)
2. When using `pocketbase update` to upgrade and replace instance, what will happen to [`custom event hooks, real-time subscriptions, and anything else that depends on the runtime state of core.App`](https://github.com/pocketbase/pocketbase/discussions/2857#discussioncomment-6364517)? The previous ones cannot be synchronized to the new instance?
3. How do you achieve zero downtime deployment on a single machine?

These questions really confuse me. I look forward to hearing your suggestions. Thank you.

**Comment by ganigeorgiev** [comment, marked as answer]
Please check https://github.com/pocketbase/pocketbase/issues/528, https://github.com/pocketbase/pocketbase/discussions/615, https://github.com/pocketbase/pocketbase/discussions/1673, https://github.com/pocketbase/pocketbase/discussions/395 and probably others.

In any case SQLite is not the reason why PocketBase can't scale horizontally and there is no option for zero downtime deployments (at least not out of the box).

`pocketbase update` downloads the latest release from GitHub and replaces the existing executable with the new one. The old application is still running and it is up to the users to restart the application process (ex. if you are using systemd that could look like `systemctl restart pocketbase`).

The Go binary startup time is usually < 1s. Existing realtime connections will be automatically reconnected in the background (this is done on the client-side).

If your use case requires HA and horizontal scaling, and downtime of 1s is an issue, then use something else.

**Reply by luminecs** [reply]
Very inspiring answer! Thank you! 

I will try to use `systemd` to deploy a new instance, and downtime of 1s is not an issue. I am planning to use pocketbase to start my new project, thank you very much for your excellent work.
